### DISCUSSION TITLE
Multiple instances/IPs/ports

### DISCUSSION THREAD

**Discussion by jonassimpson** [mikebrady/shairport-sync#1590] [Q&A]
Having read through a bunch of the discussion threads that touch on this, I feel like I'm 90% of the way to achieving what I want. That is, specifically, AirPlay2 multi-room playback via multiple instances of shairport-sync via separate sound devices. From what I understand, multiple AirPlay2 instances on the same IP doesn't work well/reliably/at all, so I'm trying to confirm if is there a way to achieve the following config:

 - assign multiple IPs to my bare-metal Linux box (miniPC running Ubuntu) (if I want six zones I can easily spare six static IPs on my home network)
 - run multiple instances of shairport-sync, each with a separate config file and listening on a different IP address
 - use the "interface" value in the conf file to have the mDNS advertising for each zone only happen on a specific interface/IP address

The part that is unclear to me is whether shairport-sync can be made to bind only to a specific IP address rather than listening on all IPs. Perhaps this is not necessary as long as I specify a unique port for each instance and then, even if they are listening on all IPs, connections will be sent to the port that belongs to the appropriate instance and it will all 'just work'.

Any insight would be helpful... I feel like I'm very close.

Thx,

JS

**Comment by mikebrady** [comment]
@noelhibbard has looked at this in some depth -- a search for his name in the Issues should help.
